ecs-init/cache/cache_test.go

```diff
@@ -291,10 +310,13 @@ func TestDownloadAgentMD5Mismatch(t *testing.T) {
 	mockFS.EXPECT().TeeReader(tarballResponse.Body, gomock.Any())
 	mockFS.EXPECT().Copy(tempfile, gomock.Any()).Return(int64(0), nil)
 	mockFS.EXPECT().Remove(tempfile.Name())
+	mockMetadata := NewMockinstanceMetadata(mockCtrl)
+	mockMetadata.EXPECT().Region().Return(config.DefaultRegionName, nil)
```

Comment:
Same as above. Can we be more strict and the check the ordering here (Using [`gomock.InOrder`](https://godoc.org/github.com/golang/mock/gomock#InOrder))?